docs/src/architecture/notes/note_lifecycle.md

```diff
@@ -0,0 +1,11 @@
+# Note lifecycle
```

Comment:
I'm not sure I'd have this section (at least in its current form). We can probably integrate this content in other places though. Also, I think we could have a simple diagram illustrating the flow (in a simplified form).

Reply:
I changed the section